### PR TITLE
Support python 3.7

### PR DESCRIPTION
**Pull request by erseco**
## Description
Added an option to install Zappa in python 3.7 because is the official python3 installed via homebrew in MacOS

### CODE REVIEW

**Comment by cclauss** [comment]
There is a bigger issue here...  __async__ is a reserved word in Python >= 3.7 which raises syntax errors.

[flake8](http://flake8.pycqa.org) testing of https://github.com/Miserlou/Zappa on Python 3.7.0

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./tests/tests_async.py:12:16: E999 SyntaxError: invalid syntax
from zappa.async import AsyncException, LambdaAsyncResponse, SnsAsyncResponse
               ^
./tests/test_app.py:2:16: E999 SyntaxError: invalid syntax
from zappa.async import task
               ^
2     E999 SyntaxError: invalid syntax
2
```

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ language: python
 python:
   - "2.7"
   - "3.6"
+  - "3.7"
```
Comment (cclauss) [inline .travis.yml]: This won't work without sudo and xenial https://github.com/travis-ci/travis-ci/issues/9069

**Comment by NorthIsUp** [comment]
3.7 is now a supported runtime on lambda.

https://aws.amazon.com/blogs/compute/python-3-7-runtime-now-available-in-aws-lambda/

**Comment by cclauss** [comment]
Please rebase and resolve conflicts.

Given that __from zappa.async ...__ is a syntax error in Python >= 3.7, should we follow the CUDA example and rename __https://github.com/Miserlou/Zappa/blob/master/zappa/async.py__ —> 
__https://github.com/Miserlou/Zappa/blob/master/zappa/non_blocking.py__?

**Comment by jgissend10** [comment]
These aren't the only changes that need to occur to allow deploying as python 3.7 are they? I see other references to python 2.7/3.6 in the code base.

**Comment by jgissend10** [comment]
> Please rebase and resolve conflicts.
> 
> Given that **from zappa.async ...** is a syntax error in Python >= 3.7, should we follow the CUDA example and rename **https://github.com/Miserlou/Zappa/blob/master/zappa/async.py** —>
> **https://github.com/Miserlou/Zappa/blob/master/zappa/non_blocking.py**?

https://github.com/Miserlou/Zappa/issues/1666

**Comment by willgdjones** [comment]
Looking forward to seeing this!

**Comment by purificant** [comment]
Proposed patch: https://github.com/Miserlou/Zappa/pull/1762

File: zappa/__init__.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import sys
 
-SUPPORTED_VERSIONS = [(2, 7), (3, 6)]
+SUPPORTED_VERSIONS = [(2, 7), (3, 6), (3, 7)]
```
Comment (cclauss) [inline zappa/__init__.py]:
```python
if sys.version_info[:2] not in SUPPORTED_VERSIONS:
    formatted_supported_versions = ['{}.{}'.format(mav, miv) for mav, miv in SUPPORTED_VERSIONS]
    err_msg = ('This version of Python ({}.{}) is not supported!\n'.format(*sys.version_info) +
               'Zappa (and AWS Lambda) support the following versions of Python: {}'.format(formatted_supported_versions))
    raise RuntimeError(err_msg)
```

**Comment by erseco** [comment]
Closed because someone is working on this #1762 to solve the use of `async` as reserved work

**Pull request closed without merging**